client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,339 @@
+
+    order = pyblish.api.CollectorOrder + 0.420
+    label = "Collect Deadline JobInfo"
+
+    families = FARM_FAMILIES
+    targets = ["local"]
+
+    profiles = []
+    pool_enum_values = []
+    group_enum_values = []
+    limit_group_enum_values = []
+    machines_enum_values = []
+
+    def process(self, instance):
+        if not instance.data.get("farm"):
+            self.log.debug("Should not be processed on farm, skipping.")
+            return
+
+        attr_values = self._get_jobinfo_defaults(instance)
+
+        attr_values.update(self.get_attr_values_from_data(instance.data))
+        job_info = AYONDeadlineJobInfo.from_dict(attr_values)
+
+        self._handle_machine_list(attr_values, job_info)
+
+        self._handle_additional_jobinfo(attr_values, job_info)
+
+        instance.data["deadline"]["job_info"] = job_info
+
+        # pass through explicitly key and values for PluginInfo
+        plugin_info_data = None
+        if attr_values["additional_plugin_info"]:
+            plugin_info_data = (
+                json.loads(attr_values["additional_plugin_info"]))
+        instance.data["deadline"]["plugin_info_data"] = plugin_info_data
+
+    def _handle_additional_jobinfo(self,attr_values, job_info):
+        """Adds not explicitly implemented fields by values from Settings."""
+        additional_job_info = attr_values["additional_job_info"]
+        if not additional_job_info:
+            return
+        for key, value in json.loads(additional_job_info).items():
+            setattr(job_info, key, value)
+
+    def _handle_machine_list(self, attr_values, job_info):
+        machine_list = attr_values["machine_list"]
+        if machine_list:
+            if attr_values["machine_list_deny"]:
+                job_info.Blacklist = machine_list
+            else:
+                job_info.Whitelist = machine_list
+
+    @classmethod
+    def apply_settings(cls, project_settings):
+        settings = project_settings["deadline"]
+        profiles = settings["publish"][cls.__name__]["profiles"]
+
+        cls.profiles = profiles or []
+
+        addons_manager = AddonsManager(project_settings)
+        deadline_addon = addons_manager["deadline"]
+        deadline_server_name = settings["deadline_server"]
+        pools = deadline_addon.get_pools_by_server_name(deadline_server_name)
+        for pool in pools:
+            cls.pool_enum_values.append({"value": pool, "label": pool})
+
+        groups = deadline_addon.get_groups_by_server_name(deadline_server_name)
+        for group in groups:
+            cls.group_enum_values.append({"value": group, "label": group})
+
+        limit_groups = (
+            deadline_addon.get_limit_groups_by_server_name(deadline_server_name))
+        if not limit_groups:
+            limit_groups.append("none")  # enum cannot be empty
+        for limit_group in limit_groups:
+            cls.limit_group_enum_values.append(
+                {"value": limit_group, "label": limit_group})
```

Comment:
```suggestion
        cls.limit_group_enum_values = [
            {"value": limit_group, "label": limit_group}
            for limit_group in limit_groups
        ]
```

But I would preffer this:
```python
        limit_groups = deadline_addon.get_limit_groups_by_server_name(
            deadline_server_name
        )
        limit_group_items = [
            {"value": limit_group, "label": limit_group}
            for limit_group in limit_groups
        ]
        if not limit_group_items:
            limit_group_items.append({"value": None, "label": "< none >"})
        cls.group_enum_values = limit_group_items
```